[BUG] Intermittent CI failure: test_deepseek_nsa_gqa_window_sliding not close (max err ~5.56)

## Describe the Bug

CI occasionally fails on `tests/ops/test_deepseek_nsa_gqa_window_sliding.py::test_nsa_gqa_window_sliding_op` with a large mismatch between the TileOPs kernel output and the PyTorch reference.

Failing example (from CI):

```
FAILED tests/ops/test_deepseek_nsa_gqa_window_sliding.py::test_nsa_gqa_window_sliding_op[3-16-8192-8192-64-128-True-2048-0-dtype1-accum_dtype1-False] - AssertionError: not close, max err: 5.55859375
```

The test uses random `cu_seqlens_q/cu_seqlens_k` and random `q/k/v` without a fixed seed, so CI flakiness appears when certain random partitions trigger the issue.

## To Reproduce

Run the failing test repeatedly (it is randomized):

```bash
pytest -q tests/ops/test_deepseek_nsa_gqa_window_sliding.py -k nsa_gqa_window_sliding -vv
# optionally loop a few times
```

Parameters observed to fail:
- `batch_size=3, groups=16, uq=8192, ukv=8192, heads=64, dim=128`
- `is_causal=True, window_size_left=2048, window_size_right=0`
- `dtype=float16, accum_dtype=float32`

## Expected Behavior

Kernel output should match the PyTorch reference within `atol=3e-3, rtol=1e-5` for all valid random partitions.

## Additional Context / Suspicion

The TileLang kernel (`tileops/kernels/deepseek_nsa/gqa_window_sliding.py`) performs full-block `T.copy(...)` loads for `q/k/v` even for partial blocks at the end of a sequence. For varlen unpadded layouts, the last batch segment ends exactly at `uq/ukv`, so reading a full block past the segment can become a true OOB global memory read (not just “reading next sequence”), which may lead to intermittent silent corruption / wrong outputs.

A robust fix likely needs bounded/partial copies (fill shared with 0, then copy only the valid rows) and/or guarding blocks where `bx*block_m >= q_current_seqlen` to avoid executing unsafe loads.

--- a/tileops/kernels/deepseek_nsa/gqa_window_sliding.py
+++ b/tileops/kernels/deepseek_nsa/gqa_window_sliding.py
@@ -85,7 +85,7 @@ def _gqa_window_sliding_kernel(
                 offset = kv_current_seqlen - q_current_seqlen
 
                 if is_causal:
-                    max_visible_k_idx = offset + (bx + 1) * block_m
+                    max_visible_k_idx = T.max(offset + (bx + 1) * block_m, 0)
                     loop_range = T.min(
                         T.ceildiv(max_visible_k_idx, block_n),
                         T.ceildiv(kv_current_seqlen, block_n))

